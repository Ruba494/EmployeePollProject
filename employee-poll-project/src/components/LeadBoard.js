import { useEffect, useState } from "react";
import { connect } from "react-redux";
import Nav from "./Nav";

function Leaderboard({ users, authedUser }) {
  const [leaderboard, setLeaderboard] = useState([]);
  useEffect(() => {
    setLeaderboard(
      users
        .map((user) => {
          const answers = Object.keys(user.answers);
          return {
            name: user.name,
            avatarURL: user.avatarURL,
            answers: answers.length,
            questions: user.questions.length,
            total: answers.length + user.questions.length,
          };
        })
        .sort((a, b) => b.total - a.total)
    );
  }, [users, authedUser]);

  return (
    <div>
      <div>
        <Nav />
      </div>
      <div className='leaderboard-container'>
        <div className='leaderboard-header'>
          <h2>Leaderboard</h2>
        </div>
        <div className='leaderboard-body'>
          <table className='Leaderboard-table'>
            <thead>
              <tr>
                <th>Users</th>
                <th>Answered</th>
                <th>Created</th>
              </tr>
            </thead>
            <tbody>
              {leaderboard.map((user) => {
                return (
                  <tr key={user.name}>
                    <td className='user-data'>
                      <img
                        src={user.avatarURL}
                        alt={user.name}
                        className='avatar'
                      />
                      {user.name}
                    </td>
                    <td>{user.answers}</td>
                    <td>{user.questions}</td>
                  </tr>
                );
              })}
            </tbody>
          </table>
        </div>
      </div>
    </div>
  );
}

const mapStateToProps = ({ users, authedUser }) => {
  return {
    users: Object.keys(users).map((key) => users[key]),
    authedUser,
  };
};
export default connect(mapStateToProps)(Leaderboard);
