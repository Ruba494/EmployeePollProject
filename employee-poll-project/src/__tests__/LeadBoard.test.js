import React from "react";
import renderer from "react-test-renderer";
import store from "../utils/test-util";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import LeadBoard from "../components/LeadBoard";

it("should render an empty message when no LeadBoard", () => {
    const elem = renderer
        .create(
            <Provider store={store}>
                <BrowserRouter>
                    <LeadBoard />
                </BrowserRouter>
            </Provider>
        )
        .toJSON();
    expect(elem).toMatchSnapshot();
});
