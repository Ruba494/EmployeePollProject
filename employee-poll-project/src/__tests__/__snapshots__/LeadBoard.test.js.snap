// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an empty message when no LeadBoard 1`] = `
<div>
  <div
    className="leaderboard-container"
  >
    <div
      className="leaderboard-header"
    >
      <h2>
        Leaderboard
      </h2>
    </div>
    <div
      className="leaderboard-body"
    >
      <table
        className="Leaderboard-table"
      >
        <thead>
          <tr>
            <th>
              Users
            </th>
            <th>
              Answered
            </th>
            <th>
              Created
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</div>
`;
